frame/babe/src/lib.rs

```diff
@@ -295,24 +295,35 @@ decl_module! {
 	}
 }
 
-mod weight {
+mod weight_for {
 	use frame_support::{
 		traits::Get,
-		weights::{constants::WEIGHT_PER_MICROS, Weight},
+		weights::{
+			constants::{WEIGHT_PER_MICROS, WEIGHT_PER_NANOS},
+			Weight,
+		},
 	};
 
-	pub fn weight_for_report_equivocation<T: super::Trait>() -> Weight {
+	pub fn report_equivocation<T: super::Trait>(validator_count: u32) -> Weight {
+		// we take the validator set count from the membership proof to
+		// calculate the weight but we set a floor of 100 validators.
+		let validator_count = validator_count.min(100) as u64;
```

Comment:
this would be a ceiling of 100.

```suggestion
		let validator_count = validator_count.max(100) as u64;
```

Reply:
Good catch! Added a basic sanity check test for this weight calculation.